broadinstitute/single_cell_portal_core
# Enable users to filter genes by differential expression metrics (SCP-5090)

This enables users to find genes by significance and size of differential expression.

Previously, genes could only be _sorted_ by these differential expression metrics.  Now, DE genes can also be _filtered_ to exclude results that aren't significant, or have low levels or differential expression, or both.  New range slider facets let users filter by those metrics.  This satisfies a stakeholder request to enable ranking genes by log<sub>2</sub>(fold change) after applying a <= 0.05 threshold for adjusted p-value.

Here's how it looks!

https://github.com/broadinstitute/single_cell_portal_core/assets/1334561/287cde62-53f5-4a51-b8d2-834ff3889e93

In addition to layout shown above, the facets are also somewhat responsive to different viewport widths.  Analytics on changes to the range facets are logged to Mixpanel.

### Test
Enhanced automated tests verify some range facet functionality.  To manually test:

* Go to "Human milk" DE pilot study
* Select annotation "cell_type__ontology_label"
* Click "Differential expression" button
* Change range slider facets for "log<sub>2</sub>(fold change)" and "Adj. p-value" to filter for different ranges
* Confirm DE table changes, either in DE genes shown on page in view or in page count at bottom
* Click checkbox beside range facet, confirm it makes the facet inactive -- i.e. doesn't apply any filters in that facet
* _If you have author DE populated for the "General_Celltype" annotation:_
* Select annotation "General_Celltype"
* Confirm range facet "Adj. p-value" does not appear, and that "q-value" does instead
* Change facets, confirm shown DE genes satisfy them

This satisfies SCP-5090.
